CRM/Contact/Form/Contact.php

```diff
@@ -704,8 +704,9 @@ public static function formRule($fields, &$errors, $contactId, $contactType) {
       }
     }
 
-    // Check for duplicate contact if it wasn't already handled by ajax or disabled
-    if (!Civi::settings()->get('contact_ajax_check_similar') || !empty($fields['_qf_Contact_refresh_dedupe'])) {
+    // Check for duplicate contact if it isn't disabled
+    $checkSimilar = Civi::settings()->get('contact_ajax_check_similar');
```

Comment:
Ouch - that's awful! We should probably use name options or a function eg

```
  'options' => ['1' => ts('While Typing'), '0' => ts('When Saving'), '2' => ts('Never')],

vs

  'options' => ['on_type' => ts('While Typing'), 'on_save' => ts('When Saving'), 'never' => ts('Never')],
  
```

The function is probably better / would avoid having to convert them. I'm just gonna try to recall what I hit in the past on that

Reply:
I think there should be two settings.  One to enable/disable the search for matching contacts upon changing a field.  And the other to enable/disable duplicate checking during validation.